mypy/options.py

```diff
@@ -177,17 +180,22 @@ def __repr__(self) -> str:
         return 'Options({})'.format(pprint.pformat(self.__dict__))
 
     def clone_for_module(self, module: str) -> 'Options':
+        res = self.clone_cache.get(module)
```

Comment:
Add docstring that mentions that after this method has been called the options object should be treated as immutable, because otherwise the cache would become stale.